models.rb

```diff
@@ -15,6 +15,16 @@ def validate
     super
     errors.add(:name, 'cannot be empty') if !name || name.empty?
   end
+
+  def remove(client)
+    unless client.projects.empty?
```

Comment:
I think you can refer to the current instance as `self` like in Django models, so you could say `unless self.projects.empty?`